src/codemagic/tools/keychain.py

```diff
@@ -273,13 +320,18 @@ def _add_certificate(self,
             certificate_password = Password('')
             obfuscate_patterns = []
 
-        process = self.execute([
+        import_cmd = [
             'security', 'import', certificate_path,
             '-f', "pkcs12",
             '-k', self.path,
-            '-T', shutil.which('codesign'),
             '-P', certificate_password.value,
-        ], obfuscate_patterns=obfuscate_patterns)
+        ]
+        if allow_for_all_apps:
+            import_cmd.append('-A')
+        for allowed_application in allowed_applications:
+            import_cmd.extend(['-T', str(allowed_application)])
```

Comment:
maybe change `str(allowed_applications)` to `shutil.which(allowed_applications)` just in case

Reply:
With changes introduced in 8b8e19bb77e7a847662049d132a8df97ffaf826f all given application paths are either fully resolved, or an exception is thrown in case they are not found from path or they don't exist.